src/sdk/namespace_fs.js

```diff
@@ -162,6 +166,14 @@ class NamespaceFS {
             params.xattr_copy; // TODO, DO we need to hard link at TaggingDirective 'REPLACE'?
     }
 
+    // get a stream and returns it's md5_b64
+    get_stream_md5_b64(stream) {
+        const md5_stream = stream_utils.get_stream_md5();
+        stream.pipe(md5_stream);
+        const md5_b64 = md5_stream.md5.digest('base64');
```

Comment:
This cannot work - the digest is not yet known at the time of setting up the pipe.
Only once the entire stream pipes through, then it will have the digest ready.
In any case, as we discussed, we don't want to run the hash in main thread.